base/deferred_task.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "std/thread.hpp"
+#include "std/chrono.hpp"
+#include "std/mutex.hpp"
+#include "std/condition_variable.hpp"
+#include "std/function.hpp"
+
+
```

Comment:
Redundant blank line. Also, please put the class into "my" namespace (as other classes in this folder do).